packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -4416,6 +4416,108 @@ addEventListener('fetch', event => {});`
 			`);
 		});
 
+		it("should error when deploying an assets-only Worker with unsupported configuration for assets-only", async () => {
+			writeWranglerConfig({
+				assets: {
+					directory: "assets",
+				},
+				send_metrics: true,
+				vars: {
+					MY_VAR: "my var",
+				},
+				services: [
+					{
+						binding: "BINDING_NAME",
+						service: "WORKER_NAME",
+					},
+				],
+				kv_namespaces: [
+					{
+						id: "123",
+						binding: "KV_BINDING_NAME",
+					},
+				],
+			});
+			await expect(runWrangler("deploy")).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Assets-only Workers do not support the following configuration keys:
+
+				⋅ "send_metrics"
```

Comment:
send_metrics is a Wrangler feature which logs usage, no? So that does still feel relevant to static sites.